http/longpoll/longpoll.go

```diff
@@ -0,0 +1,75 @@
+package longpoll
+
+import (
+	"context"
+	"time"
+)
+
+// LongPollFunc should return true if long polling did
+// resolve, false otherwise. If error is returned, the
+// long polling will be canceled
+type LongPollFunc func() (bool, error)
+
+// Config for long polling
+type Config struct {
+	// RetryTime time to wait between two retries
+	RetryTime time.Duration
+	// MinWaitTime min time to wait
+	MinWaitTime time.Duration
+	// MaxWaitTime max time to wait
+	MaxWaitTime time.Duration
+}
+
+// Default configuration for http long polling
+// wait half a second between retries, min 1 sec and max 60 sec
+var Default = Config{
+	RetryTime:   time.Millisecond * 500,
+	MinWaitTime: time.Second,
+	MaxWaitTime: time.Second * 60,
+}
+
+// Until executes the given function fn until duration d is passed or context is canceled
+func Until(ctx context.Context, d time.Duration, fn LongPollFunc) (ok bool, err error) {
+	return Default.LongPollUntil(ctx, d, fn)
+}
+
+// LongPollUntil executes the given function fn until duration d is passed or context is canceled
+func (c Config) LongPollUntil(ctx context.Context, d time.Duration, fn LongPollFunc) (ok bool, err error) {
+	until := time.Now()
+	if d >= c.MinWaitTime && d < c.MaxWaitTime {
```

Comment:
I think returning an error in case `d` is out of the min/max range would make the behaviour more clear to the caller. If I understand correctly it would just silently perform a single request if `d` is bigger than `c.MaxWaitTime` or smaller than `c.MinWaitTime`.

Reply:
Why even restrict this?

Reply:
I don't like to handle an error in this case but I agree, that the caller might not understand the behavior. It needs to be restricted to avoid long hanging go-routines in http handlers and the like. So instead of dropping the arg silently the value will be set to min/max if duration is given and otherwise out of bounds

Reply:
Please also update the comment of both functions that there are boundaries and that this is the behaviour.

Reply:
Yes, did so.